### PR TITLE
terragrunt-info parsing improvement

### LINKED ISSUE #2885: terragrunt-info does not work with mock outputs
**Describe the bug**
terragrunt-info fails to work with mock outputs of a dependency

**To Reproduce**

Module A

```hcl
# config/module-a/terragrunt.hcl
terraform {
  source = ".//"
}

# config/module-a/main.tf
output "test_var" {
  value = "hello"
} 
```

Module B
```hcl
# config/module-b/terragrunt.hcl
terraform {
  source = ".//"
}

dependency "module_a" {
  config_path                             = "../module-a"
  mock_outputs_allowed_terraform_commands = ["init", "plan", "validate"]
  mock_outputs_merge_strategy_with_state  = "shallow"
  mock_outputs = {
    test_mock = "abc"
  }
}

inputs = {
  test_var = dependency.module_a.outputs.test_mock
}

# config/module-b/main.tf
variable "test_var" {
  type = string
}

output "result" {
  value = var.test_var
}
```

Working:
`cd config/module-b && terragrunt plan`

```text
INFO[0000] Downloading Terraform configurations from file:///my-home/terragrunt-info-test/config/module-a into /my-home/terraform.d/.terragrunt-cache/AnwNMwh9yg0XKSaMbmsxdNYr1BI/o12cAWD2nsVQ_xmslaEWdI5j6Fs  prefix=[/my-home/terragrunt-info-test/config/module-a] 

Initializing the backend...

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

Changes to Outputs:
  + result = "abc"

You can apply this plan to save these new output values to the Terraform
state, without changing any real infrastructure.

─────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't
guarantee to take exactly these actions if you run "terraform apply" now.
```

Failing:
`cd config/module-b && terragrunt terragrunt-info`

```text
terragrunt terragrunt-info 

ERRO[0000] /my-home/terragrunt-info-test/config/module-a/terragrunt.hcl is a dependency of /my-home/Developer/oss/terragrunt-info-test/config/module-b/terragrunt.hcl but detected no outputs. Either the target module has not been applied yet, or the module has no outputs. If this is expected, set the skip_outputs flag to true on the dependency block. 
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1 
```

**Expected behavior**
terragrunt-info should use the mock outputs when running

**Versions**
- Terragrunt version: 0.54.12
- Terraform version: 1.6
- Environment details (Ubuntu 20.04, Windows 10, etc.):

**Additional context**
The same happens with an initialised and applied module where the output is new and has not yet been applied and saved to state. This will result in the outputs object not containing any mock outputs.

### PR DESCRIPTION
**Pull request by denis256**
## Description

Added returning of `terragrunt-info` output in case of error.

Before this change:
```
INFO[0001] Downloading Terraform configurations from file:///projects/gruntwork/terragrunt-tests/issue-2885/config/module-a into /projects/gruntwork/terragrunt-tests/issue-2885/config/module-a/.terragrunt-cache/75aBMnP5LlsdDf8XAME_pjiHnKg/c4EpdguDY52bRbC94e0VP6SzGqA  prefix=[/projects/gruntwork/terragrunt-tests/issue-2885/config/module-a] 
ERRO[0001] /projects/gruntwork/terragrunt-tests/issue-2885/config/module-a/terragrunt.hcl is a dependency of /projects/gruntwork/terragrunt-tests/issue-2885/config/module-b/terragrunt.hcl but detected no outputs. Either the target module has not been applied yet, or the module has no outputs. If this is expected, set the skip_outputs flag to true on the dependency block. 
ERRO[0001] Unable to determine underlying exit code, so Terragrunt will exit with error code 1 
```
With changes from this PR:
```
$ terragrunt-2901 terragrunt-info
{
  "ConfigPath": "/projects/gruntwork/terragrunt-tests/issue-2885/config/module-b/terragrunt.hcl",
  "DownloadDir": "/projects/gruntwork/terragrunt-tests/issue-2885/config/module-b/.terragrunt-cache",
  "IamRole": "",
  "TerraformBinary": "terraform",
  "TerraformCommand": "terragrunt-info",
  "WorkingDir": "/projects/gruntwork/terragrunt-tests/issue-2885/config/module-b"
}
ERRO[0000] /projects/gruntwork/terragrunt-tests/issue-2885/config/module-a/terragrunt.hcl is a dependency of /projects/gruntwork/terragrunt-tests/issue-2885/config/module-b/terragrunt.hcl but detected no outputs. Either the target module has not been applied yet, or the module has no outputs. If this is expected, set the skip_outputs flag to true on the dependency block. 
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

Fixes #2885.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

Updated `terragrunt-info` to return parsed info data in case of error.

### CODE REVIEW

**Comment by caspar-ds** [comment]
Hello! Any ideas on when this might end up being merged? Any I can do to help out? Thanks

**Comment by caspar-ds** [comment]
Hi @denis256 and thanks for the PR that addresses the issue! Any ideas on when this might end up being merged? Anything I can do to help out? Thanks

**Comment by caspar-ds** [comment]
Hi @denis256 - thanks again! Anything I can do to help with this PR?

**Review by caspar-ds — APPROVED** [review]

File: cli/commands/terragrunt-info/action.go [hunk 1]
```diff
@@ -40,8 +40,20 @@ func runTerragruntInfo(opts *options.TerragruntOptions, cfg *config.TerragruntCo
 		opts.Logger.Errorf("JSON error marshalling terragrunt-info")
 		return err
 	}
-	fmt.Fprintf(opts.Writer, "%s\n", b)
-
+	if _, err := fmt.Fprintf(opts.Writer, "%s\n", b); err != nil {
+		return err
```
Comment (levkohimins) [inline cli/commands/terragrunt-info/action.go:44]:
```suggestion
		return errors.WithStackTrace(err)
```

File: cli/commands/terragrunt-info/action.go [hunk 2]
```diff
@@ -40,8 +40,20 @@ func runTerragruntInfo(opts *options.TerragruntOptions, cfg *config.TerragruntCo
 		opts.Logger.Errorf("JSON error marshalling terragrunt-info")
 		return err
```
Comment (levkohimins) [inline cli/commands/terragrunt-info/action.go:41]:
```suggestion
		return errors.WithStackTrace(err)
```

**Review by levkohimins — CHANGES_REQUESTED** [review]
Fdded a few small suggestions

**Review by levkohimins — APPROVED** [review]
LGTM!

**Pull request merged**